### PR TITLE
Angular 13.0.0 working version

### PR DESCRIPTION
**Pull request by stefanoslig**
Issue:

## What I did
Some tweaks in this branch([angular/update-13](https://github.com/storybookjs/storybook/tree/angular/update-13)) so we can use Angular 13 and storybook

image

## How to test

- [ ] Is this testable with Jest or Chromatic screenshots?
- [ ] Does this need a new example in the kitchen sink apps?
- [ ] Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

cc @shilman , @ThibaudAV

### CODE REVIEW

File: app/angular/src/builders/start-storybook/index.ts [hunk 1]
```diff
@@ -40,7 +40,7 @@ export type StorybookBuilderOptions = JsonObject & {
 
 export type StorybookBuilderOutput = JsonObject & BuilderOutput & {};
 
-export default createBuilder(commandBuilder);
+export default createBuilder<any>(commandBuilder);
```
Comment (stefanoslig) [inline app/angular/src/builders/start-storybook/index.ts:43]: maybe I can use a different type here. However, `StorybookBuilderOptions` was not working

**Comment by literalpie** [comment]
Hi, what are next steps on this?
* Do these changes cause sb to work in an ng13 project? (since the problem is ESM stuff, I'm not sure if the examples project within the storybook repo will fully test the problem is fixed)
* If this does fix the esm issue, could a smaller PR address that issue without updating the example project? Or is it preferred that both parts are addressed at the same time?
* Are the issues [mentioned here](https://github.com/storybookjs/storybook/pull/16528#issuecomment-960757194) fixed somehow, or were they not necessary to get ng13 working? specifically, the issues of using private Angular APIs?
* Is there any area where community help would be useful, or is there a plan in progress and we just need to be patient :-)?

**Comment by stefanoslig** [comment]
I'm closing this PR just to keep the focus on [this one]( https://github.com/storybookjs/storybook/pull/16528) 
This PR solves some problems however I was not aware that in storybook, we cannot move to a Typescript version > 3.9. This issue should be solved first and then all the others

**Comment by juristr** [comment]
Hey, we're from [Nx](https://nx.dev) are also currently in the process of updating all our tooling to make sure it works with Angular 13. Storybook is a blocker there atm :)

I tested this PR quickly today morning on an Nx 13 repo with the latest Angular 13 & Storybook and it looks good. It indeed fixes the ESM issue. I'm now doing some more extensive testing.

That said, I'm also curious for a timeline as we plan to release Nx with Angular 13 support soonish ..in like very soon 😅 . And we want Storybook to work there 🙂

**Pull request closed without merging**